apps/dashboard/app/models/concerns/job_logger.rb

```diff
@@ -0,0 +1,50 @@
+
+module JobLogger
+
+  def upsert_job!(directory, job)
+    existing_jobs = jobs(directory)
+    stored_job = existing_jobs.detect { |j| j.id == job.id && j.cluster == job.cluster }
+
+    if stored_job.nil?
+      new_jobs = (existing_jobs + [job.to_h]).map(&:to_h)
+    else
```

Comment:
It looks like completed jobs are not being saved with clusters, which is causing there to be duplicate entries in the job log that appear to never load in the projects#show page.

Reply:
Thanks for checking! I thought there was something buggy here but couldn't quite put my finger on it.